micrometer-core/src/main/java/io/micrometer/core/instrument/step/StepMeterRegistry.java

```diff
@@ -48,7 +57,11 @@ public StepMeterRegistry(StepRegistryConfig config, Clock clock) {
     }
 
     public void start() {
-        start(Executors.defaultThreadFactory());
+        start((r) -> {
```

Comment:
@dhaval24 This change looks irrelevant to this and you’ve already raised #814 and #815 , so it’d be good to be reverted here.

Reply:
@izeye we can either merge #815 before and then merge this one, or merge this before #815 and abonden it. I think it would produce the same result. Whatever is preferred I am fine with it.

Reply:
@dhaval24 IMHO, any PR would be good to focus on it own purpose in general.

Reply:
done